lib/Sema/CSSyntacticElement.cpp

```diff
@@ -954,6 +988,10 @@ class SyntacticElementConstraintGenerator
                       /*contextualInfo=*/{}, isDiscarded));
     }
 
+    if (!hadError)
+      addResultDefault(
+          cast<ClosureExpr>(context.getAsAbstractClosureExpr().getPtrOrNull()));
```

Comment:
At this spot we cannot assume that what we are dealing with is a closure, it could be a function too for example of a TapExpr.